tests/e2e/taqueria-cli-permissions.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { exec as exec1 } from 'child_process';
+import fsPromises from 'fs/promises';
+import os from 'os';
+import util from 'util';
+import { generateTestProject } from './utils/utils';
+const exec = util.promisify(exec1);
+
+const taqueriaProjectPath = './e2e/auto-test-cli-permissions';
+let username = os.userInfo().username;
+let userGroup: string;
+
+describe('E2E Testing for taqueria plugin file permissions,', () => {
+	beforeAll(async () => {
+		await generateTestProject(taqueriaProjectPath, ['ligo', 'archetype', 'smartpy', 'contract-types']);
+
+		await exec(`cp e2e/data/fa12.arl ${taqueriaProjectPath}/contracts`);
+		await exec(`cp e2e/data/increment.jsligo ${taqueriaProjectPath}/contracts`);
+		await exec(`cp e2e/data/hello-tacos.py ${taqueriaProjectPath}/contracts`);
+
+		userGroup = (await exec(`id -g -n ${username}`)).stdout.trim();
+	});
+
+	test('testing that ligo artifacts will have the correct permissions', async () => {
+		await exec(`taq compile --plugin ligo`, { cwd: `./${taqueriaProjectPath}` });
+		const fileUser = await exec(`stat -c %U ${taqueriaProjectPath}/artifacts/increment.tz`);
+		const fileGroup = await exec(`stat -c %G ${taqueriaProjectPath}/artifacts/increment.tz`);
+
+		expect(fileUser.stdout.trim()).toBe(username);
+		expect(fileGroup.stdout.trim()).toBe(userGroup);
```

Comment:
If the user running these tests is `root`, (Is it? in our CI pipelines?) then these tests will pass anyway, even if the functionality being tested sets permissions to `root` not to the current user.
TBH, I don't have a suggestion to fix this.